testing/resource.go

```diff
@@ -65,11 +66,34 @@ func (c *Resource) SetDefaults(ctx context.Context) {
 	c.Spec.SetDefaults(ctx)
 }
 
+type onContextKey struct{}
+
+// WithContext is the type of a context decorator.
+type WithContext func(context.Context) context.Context
```

Comment:
I would call this `MutateContext`

Reply:
I removed this based on the below comment.